Assets/HoloToolkit/Utilities/Scripts/Extensions/StaticExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// This is a set of Static Extensions for determining if an object is a prefab or in the scene
+    /// </summary>
+    static class StaticExtensions 
+    {
+        /// <summary>
+        /// Function to check at runtime if the transform is an instance or a prefab
+        /// </summary>
+        /// <param name="This">
+        /// A <see cref="Transform"/> for the object to validate as prefab.
+        /// </param>
+        internal static bool IsPrefab(GameObject go)
```

Comment:
Why do we need this?

Reply:
There's no way to tell if a linked game object is either a prefab or a game object placed in the scene.  For the ObjectButton I allow the user to link to game objects that are either in the scene or a prefab and if it's a prefab I spawn it vs. referencing it directly.

Reply:
This is untrue. That's what [`PrefabUtility`](https://docs.unity3d.com/ScriptReference/PrefabUtility.html) is for.

Reply:
What about using `gameObject.Scene` 
![image](https://user-images.githubusercontent.com/5775094/31293627-0b6b8452-aad8-11e7-96f2-18a91e615816.png)

Reply:
Yeah @Zod is correct that's why I wrote the class so I could do it runtime.  I've never looked at doing gameObject.scene to validate.  I'll swap it out in the ObjectButton and remove the extension.  Much simpler.

thanks,
-pat

Reply:
Why do you need to know if it's a prefab at runtime?